opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1260,6 +1255,28 @@ void CpGrid::markElemAssignLevelDetectActiveLgrs(const std::vector<std::array<in
             if(belongsToLevel) {
                 this-> mark(1, element);
                 assignRefinedLevel[element.index()] = level+1; // shifted since starting grid is level 0, and refined grids levels are >= 1.
+            }
+        }
+    }
+}
+
+void CpGrid::detectActiveLgrs(const std::vector<std::array<int,3>>& startIJK_vec,
```

Comment:
Good point! The reason is that in adapt(/* with arguments */)  we only call detectActiveLgrs(...) since the elements get marked somewhere else. In addLgrsUpdateLeafView(...), indeed we call both methods, one after the other. 

I'll incorporate the idea of a std::function/lambda to reduce code duplication. Many thanks!


To do: change the name of adapt(/* with arguments */) to something else, not to confuse with DUNE's grid interface adapt().